Voycar.Api.Web.Tests.Integration/Users/Post_ResetPassword/Tests.cs

```diff
@@ -0,0 +1,168 @@
+    }
+
+
+    [Fact]
+    public async Task Post_Request_AsAdmin_ReturnsOk_And_ResetPassword()
+    {
+        // Arrange
+        const string email = "admin@test.de";
+        var adminClient = await ClientFactory.CreateAdminClient(this._app, this.Context, email, "password");
+        var request = new R.Request
+        {
+            PasswordResetToken = "799A354C-7AD6-4305-8F34-72DC4ABFDD2F", Password = "newPassword"
+        };
+
+        var userInDb = this.Context.Users.First(user => user.Email == email);
+        userInDb.PasswordResetToken = "799A354C-7AD6-4305-8F34-72DC4ABFDD2F";
+        userInDb.ResetTokenExpires = DateTime.UtcNow.AddHours(3);
+
+        await this.Context.SaveChangesAsync();
+
+        // Act
+        var firstHttpResponse = await adminClient.POSTAsync<R.Endpoint, R.Request>(request);
+        var secondHttpResponse = await adminClient
+            .POSTAsync<L.Endpoint, L.Request>(new L.Request { Email = email, Password = request.Password });
+
+        // Assert
+        userInDb.Should().NotBeNull();
+
+        firstHttpResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+        secondHttpResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+    }
+
+
+    [Fact]
+    public async Task Post_Request_ReturnsBadRequest_DueToExpiredResetToken()
+    {
+        // Arrange
+        const string email = "member@test.de";
+        var memberClient = await ClientFactory.CreateMemberClient(this._app, this.Context, email, "password");
+        var request = new R.Request
+        {
+            PasswordResetToken = "F01E0980-AB43-47E2-9300-DA69838E8797", Password = "newPassword"
+        };
+
+        var userInDb = await this.Context.Users.FirstOrDefaultAsync(user => user.Email == email);
+        userInDb.PasswordResetToken = "F01E0980-AB43-47E2-9300-DA69838E8797";
+        userInDb.ResetTokenExpires = DateTime.UtcNow.AddHours(-12);
+        await this.Context.SaveChangesAsync();
+
+
+        // Act
+        var httpResponse = await memberClient.POSTAsync<R.Endpoint, R.Request>(request);
+
+
+        // Assert
+        userInDb.Should().NotBeNull();
+        httpResponse.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+    }
+
+
+    // Validator-Tests
+    [Fact]
+    public async Task Post_Request_ReturnsBadRequest_DueToInvalidResetToken()
+    {
+        // Arrange
+        var request = new R.Request { PasswordResetToken = null!, Password = "password" };
+
+        // Act
+        var httpResponse = await this._app.Member.POSTAsync<R.Endpoint, R.Request>(request);
+
+        // Assert
+        httpResponse.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+    }
+
+
+    [Fact]
+    public async Task Post_Request_ReturnsBadRequest_DueToInvalidPassword()
+    {
+        // Arrange
+        var request = new R.Request { PasswordResetToken = "F01E0980-AB43-47E2-9300-DA69838E8797", Password = "" };
+
+        // Act
+        var httpResponse = await this._app.Member.POSTAsync<R.Endpoint, R.Request>(request);
```

Comment:
Hier sollten wir dann aber auch sicherstellen, dass der Token gültig ist. Ja der Validator kommt immer bevor der Token überprüft wird, aber wenn der Validator nicht funktioniert, würden wir das in diesem Test gar nicht merken.
```suggestion
        // Arrange
       const string email = "member3@test.de";
        var memberClient = await ClientFactory.CreateMemberClient(this._app, this.Context, email, "password");
        var request = new R.Request
        {
            PasswordResetToken = "F01E0980-AB43-47E2-9300-DA69838E8797", Password = ""
        };

        var userInDb = this.Context.Users.First(user => user.Email == email);
        userInDb.PasswordResetToken = request.PasswordResetToken;
        userInDb.ResetTokenExpires = DateTime.UtcNow.AddHours(3);

        await this.Context.SaveChangesAsync();

        // Act
        var httpResponse = await memberClient.POSTAsync<R.Endpoint, R.Request>(request);
```